addons/event/controllers/main.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import Controller, request, route, content_disposition
+
+
+class EventController(Controller):
+
+    @route(['''/event/<model("event.event", "[('state', 'in', ('confirm', 'done'))]"):event>/ics'''], type='http', auth="public")
+    def make_event_ics_file(self, event):
+        if not event or not event.registration_ids:
+            return request.not_found()
+        files = event._get_ics_file()
+        content = files[event.id]
```

Comment:
As `_get_ics_file` can return an empty dict if `vobject` is not imported, i'd rather check if `files[event.id]` and return a NotFound if not.